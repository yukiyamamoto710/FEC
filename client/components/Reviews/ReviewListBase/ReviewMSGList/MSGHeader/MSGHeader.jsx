import React from 'react';
import PropTypes from 'prop-types';
import Stars from '../../../Stars/Stars';

const MSGHeader = (props) => {
  const { user, date, rate } = props;
  const dateStr = date.slice(0, 10);
  const str = String(new Date(dateStr)).slice(4, 15);

  return (
    <div
      data-testid="msgHeader"
      className="msgHeader"
    >
      <Stars
        classNameForSize="msgStar"
        rate={rate}
      />
      <div>
        {user}
        ,
        {' '}
        {str}
      </div>
    </div>
  );
};

MSGHeader.propTypes = {
  user: PropTypes.string,
  date: PropTypes.string,
  rate: PropTypes.number,
};

MSGHeader.defaultProps = {
  user: '',
  date: '',
  rate: 0,
};

export default MSGHeader;
