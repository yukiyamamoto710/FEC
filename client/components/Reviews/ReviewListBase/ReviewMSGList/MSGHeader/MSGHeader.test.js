/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import MSGHeader from './MSGHeader';

afterEach(cleanup);

it('should render text', () => {
  render(
    <MSGHeader
      user="Yuki"
      date="2021-05-23T00:00:00.000Z"
      rate={1}
    />,
  );
  const test = screen.queryByText('Yuki, May 22 2021');
  expect(test).toBeInTheDocument();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <MSGHeader
      user="Yuki"
      date="2021-05-23T00:00:00.000Z"
      rate={1}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
