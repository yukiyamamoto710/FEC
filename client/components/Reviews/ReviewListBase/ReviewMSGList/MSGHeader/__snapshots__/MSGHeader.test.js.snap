// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="msgHeader"
  data-testid="msgHeader"
>
  <div
    className="ratingHead"
    data-testid="stars"
  >
    <div>
      <div
        className="msgStar fullStar"
        data-testid="button0"
        id=" 1"
        label="fullStar"
        name=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="msgStar emptyStar"
        data-testid="button1"
        id=" 2"
        label="emptyStar"
        name=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="msgStar emptyStar"
        data-testid="button2"
        id=" 3"
        label="emptyStar"
        name=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="msgStar emptyStar"
        data-testid="button3"
        id=" 4"
        label="emptyStar"
        name=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
    <div>
      <div
        className="msgStar emptyStar"
        data-testid="button4"
        id=" 5"
        label="emptyStar"
        name=""
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      />
    </div>
  </div>
  <div>
    Yuki
    ,
     
    May 22 2021
  </div>
</div>
`;
